### DISCUSSION TITLE
Programmatic Usage

### DISCUSSION THREAD

**Discussion by avin-kavish** [prisma/prisma#13787] [Q&A]
Does prisma have programmatic access for library authors?

**Comment by nurul3101** [comment]
Hey @avin-kavish 👋 
If you mean programmatic access to CLI commands: [#4703](https://github.com/prisma/prisma/issues/4703), then it isn't supported yet. Do you mean something else?

**Reply by avin-kavish** [reply]
Specifically I want to migrate, db push, format, and generate client. It doesn't have to be through CLI. It can be what the CLI invokes.

**Reply by nurul3101** [reply]
We don't have the programmatic support yet.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
